src/wmtk/MultiMeshManager.cpp

```diff
@@ -344,8 +368,299 @@ std::string MultiMeshManager::parent_to_child_map_attribute_name(long index)
+    } else {
+        return std::optional<Tuple>{};
+    }
+}
+
+std::optional<Tuple> MultiMeshManager::find_valid_tuple_from_split(
+    Mesh& my_mesh,
+    const Simplex& old_simplex,
+    const long old_simplex_gid,
+    const std::vector<Tuple>& tuple_alternatives,
+    const std::vector<std::tuple<long, std::array<long, 2>>>& split_cell_maps) const
+{
+    const Tuple& old_tuple = old_simplex.tuple();
+    const PrimitiveType primitive_type = old_simplex.primitive_type();
+
+    for (const auto& [old_cid, new_cids] : split_cell_maps) {
+        if (old_cid != old_simplex_gid) {
+            continue;
+        }
+
+        auto old_tuple_opt =
+            find_tuple_from_gid(my_mesh, primitive_type, tuple_alternatives, old_cid);
+
+        assert(old_tuple_opt.has_value());
+
+        const Tuple& old_cid_tuple = old_tuple_opt.value();
+        for (const long new_cid : new_cids) {
+            // try seeing if we get the right gid by shoving in the new face id
+            Tuple tuple(
+                wmtk::utils::TupleInspector::local_vid(old_cid_tuple),
+                wmtk::utils::TupleInspector::local_eid(old_cid_tuple),
+                wmtk::utils::TupleInspector::local_fid(old_cid_tuple),
+                new_cid,
+                my_mesh.get_cell_hash_slow(new_cid));
+
+
+            if (my_mesh.is_valid_slow(tuple) &&
+                old_simplex_gid == my_mesh.id(tuple, primitive_type)) {
+                return tuple;
+            }
+        }
+    }
+    return std::optional<Tuple>{};
+}
+
+std::optional<Tuple> MultiMeshManager::find_tuple_from_gid(
+    const Mesh& my_mesh,
+    PrimitiveType primitive_type,
+    const std::vector<Tuple>& tuples,
+    long gid)
+{
+    // spdlog::info("Finding gid {}", gid);
+    // for(const auto& t: tuples) {
+    //     spdlog::info("Found {}", my_mesh.id(t, primitive_type));
+
+    //}
+
+    auto it = std::find_if(tuples.begin(), tuples.end(), [&](const Tuple& t) -> bool {
+        return gid == my_mesh.id(t, primitive_type);
+    });
+    if (it == tuples.end()) {
+        // spdlog::info("failed to find tuple");
+        return std::optional<Tuple>{};
+    } else {
+        // spdlog::info("got tuple");
+        return *it;
+    }
+}
+long MultiMeshManager::child_global_cid(
+    const attribute::ConstAccessor<long>& parent_to_child,
+    long parent_gid)
+{
+    // look at src/wmtk/multimesh/utils/tuple_map_attribute_io.cpp to see what index global_cid gets mapped to)
+    // 5 is the size of a tuple is 5 longs, global_cid currently gets written to position 3
+    return Mesh::get_index_access(parent_to_child).vector_attribute(parent_gid)(5 + 3);
+}
+long MultiMeshManager::parent_global_cid(
+    const attribute::ConstAccessor<long>& child_to_parent,
+    long child_gid)
+{
+    // look at src/wmtk/multimesh/utils/tuple_map_attribute_io.cpp to see what index global_cid gets mapped to)
+    // 5 is the size of a tuple is 5 longs, global_cid currently gets written to position 3
```

Comment:
That sounds like it could break quite easily in the future. Maybe make it a static member variable of `MultiMeshManager`? It can still break but then it would be at least visible in the header file.